netaddr.go

```diff
@@ -67,84 +76,48 @@ var (
 	z6noz = new(intern.Value)
 )
 
-func v4Prefix(ip [4]byte, bits uint8) (IPPrefix, error) {
-	if bits > 32 {
-		return IPPrefix{}, fmt.Errorf("netaddr: prefix length %d too large for IP address family", bits)
-	}
-	skip, partial := int(bits/8), bits%8
-	if partial != 0 {
-		ip[skip] = ip[skip] & ^byte(0xff>>partial)
-		skip++
-	}
-	for i := skip; i < 4; i++ {
-		ip[i] = 0
-	}
-	return IPPrefix{IPv4(ip[0], ip[1], ip[2], ip[3]), bits}, nil
-}
-
-const (
-	// mapped4Prefix are the 12 leading bytes in a IPv4-mapped IPv6 address.
-	mapped4Prefix = "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\xff\xff"
-)
-
-// v6Loopback is the IPv6 loopback address.
-var v6Loopback = [16]byte{15: 1}
-
-func v6Prefix(ip [16]byte, bits uint8) (IPPrefix, error) {
-	if bits > 128 {
-		return IPPrefix{}, fmt.Errorf("netaddr: prefix length %d too large for IP address family", bits)
-	}
-	skip, partial := int(bits/8), bits%8
-	if partial != 0 {
-		ip[skip] = ip[skip] & ^byte(0xff>>partial)
-		skip++
-	}
-	b := ip[skip:]
-	for i := range b {
-		b[i] = 0
-	}
-	return IPPrefix{IPv6Raw(ip), bits}, nil
-}
-
 // IPv6LinkLocalAllNodes returns the IPv6 link-local all nodes multicast
 // address ff02::1.
 func IPv6LinkLocalAllNodes() IP { return IPv6Raw([16]byte{0: 0xff, 1: 0x02, 15: 0x01}) }
 
 // IPv6Unspecified returns the IPv6 unspecified address ::.
-func IPv6Unspecified() IP { return IPv6Raw([16]byte{}) }
+func IPv6Unspecified() IP { return IP{z: z6noz} }
```

Comment:
I was trying to avoid using the `z*` constants whenever possible if there was a better way to do it. I'd prefer the old way unless there's some reason. I'd hope the performance is the same?

Reply:
Not strongly. It's fine as you have it.